### ISSUE TITLE
We should use 'Units of work' instead of 'Days' for the Effort estimate labels

### ISSUE THREAD

**Issue by DinisCruz** [TeamMentor/TM_4_0_Design#376]
As per the thread at https://github.com/TeamMentor/TM_4_0_Design/issues/368#issuecomment-72527056 we all have quite different interpretations of timelines, so I think it is better if we start measuring the Effort of each issue into 'Units of work' , which is something that we can (over time) gain a better understanding of what they mean (i.e. after an issue is completed, we should also adjust the Effort, and map the reality of that it really took to fix it)

Days is such a subjective metric, since by definition one of us could have a number of issues open which multiple 'effort' values, some of which could be related (for example, for a while I had a number of Issues that I would say were off  'Unit of work effort 1'. I was eventually able to do them after I completed an Issue with 'Unit of work effort 3' done. In this cases, there were a number of issues whose solution was easy, after another issue was completed.

Basically the effort is still something that we are still getting our heads around it, and the good news is that we are actually trying to measuring it (which should allow for better visibility on what is on the pipeline). So it is better to keep it a bit vague at the moment and see what actually works in practice

**Comment by roman87** [comment]
There are different schools of thoughts on this and none of them are perfect. That is why I tried to use both in my labeling with my beer container analogy :) We probably should have a smaller unit of work that 1 day.

**Comment by roman87** [comment]
Ok, just created an Effort: Taster - 4h :) 
For reference:
Taster = 157 milliliters
Schooner = 285ml
Pint=473ml
Leter = 1000ml
Keg = ~55 liters
Cistern = 10,000 liters 

I am open to renaming it. We can do a plain and boring Fibonacci sequence.

**Comment by DinisCruz** [comment]
Actually a Fibonacci sequence would be quite cool, specially since it is also related to the https://en.wikipedia.org/wiki/Golden_ratio which is a nice design thing

**Comment by roman87** [comment]
Anyone else wants to weight in on the issue? so far we are in a disagreement if we should use Beer or Fibonacci to track work effort :)?

**Comment by luchia** [comment]
It doesn't really bother me, isn't it just a general outline of how long things should take?

**Comment by roman87** [comment]
Pretty much. This is is used more to track burn-down rate. Since we don't have statistics in github, I think it is easier to estimate work effort in days. That way we can intuitively say that we were ether correct in estimating certain effort or we were off. For example, if we say something should take 1 day and it takes us 1 week, we were wrong. Over time we will get better with estimates. With Fibonacci, the purpose is the same. Over time statistically you figure out that a work effort of 8 takes about 12 days.

**Comment by DinisCruz** [comment]
I read the Scrum book http://www.amazon.co.uk/Scrum-revolutionary-approach-deadlines-productivity-ebook/dp/B00I52D6KQ/ book over the weekend and one of the things he mentions (he guy who come up with Scrum) is that task duration and effort should never be measured in actual time (since we are all really bad at it)

What we are good at is at comparing tasks, which makes the fibonacci sequence an really good model.

So lets change the hours references to the this variation of the fibonacci sequence: 1,3,8,13,21,34 (I removed the 2 and 5). We can still use the current names, just change the hours, days and weeks references

![image](https://cloud.githubusercontent.com/assets/656739/6664343/b7b694c0-cbc6-11e4-8a43-63be1c98b697.png)

**Comment by roman87** [comment]
We discussed it before :) there are different schools and different approaches, and no one size fits all. Regardless of which one we have, we need to have a reporting mechanism which produces a burn down report.  Which basically says that historically it takes us N days to complete a Y complexity story. We don't have those tools in place :) and those tools will require a lot of overhead which we don't need to take right now.  We can still produce an average time to close of each issue for each effort. 
I personally like the time estimate in our case, simply because you can easily visualize a task taking half a day, full day or 2 days. It is much harder to visualize a 3 complexity task..  
That is why, I think, until we move to a more robust Scrum tool, with reporting everyone understands, it is easier to use time. 
If you like we can just add 1,3,8,13,21,34 complexity next to each name. But I would still consider the time estimate primary. i.e. if it takes you longer than 1/2 a day its not a 1 task.

**Comment by roman87** [comment]
btw, here is how you can visualize the complexity:

Taster - http://www.google.com/imgres?imgurl=http://www.peelepaddle.com/water_4oz.jpg&imgrefurl=http://www.peelepaddle.com/BeerTastingGlasses.html&h=125&w=98&tbnid=qCFQVBh6pmpHZM:&zoom=1&docid=DZSKUu-G3Aq1SM&ei=mAAHVabJMsHbsATfnIDIBw&tbm=isch&ved=0CCkQMygNMA0

Schooner - 
http://www.google.com/imgres?imgurl=http://ecx.images-amazon.com/images/I/31EbIhQ4ROL._SY355_.jpg&imgrefurl=http://www.amazon.com/Libbey-Schooner-Beer-Glass-21/dp/B00B4DC9QA&h=355&w=355&tbnid=BTlaxmCSBQ47YM:&zoom=1&docid=qm_O54vWf4qR1M&ei=VQAHVc7aHuzbsASvpoHQAQ&tbm=isch&ved=0CDUQMygKMAo

Pint - 
http://www.google.com/imgres?imgurl=http://thecornerbarandgrill.com/wp-content/uploads/2014/06/pintglass.gif&imgrefurl=http://thecornerbarandgrill.com/specials/pintglass/&h=1298&w=726&tbnid=yj9Bn3CWV6mtnM:&zoom=1&docid=RMoX5C9BAS58QM&ei=wAAHVa_gI_KIsQTs2IDIAQ&tbm=isch&ved=0CDsQMygKMAo

Liter - 
http://www.google.com/imgres?imgurl=http://static.zoovy.com/img/bamtar/-/products/steins/glass_dimpled_steinblank_1_liter_glass_04533_80.jpg&imgrefurl=http://www.thoughtyoumayask.com/picsbtqq/1-liter-of-water-mass&h=554&w=429&tbnid=hlp46zWABA7OPM:&zoom=1&docid=MKM_a1g1NZxjYM&ei=5gAHVdjiFYWxsATy44LwBQ&tbm=isch&ved=0CB4QMygCMAI

Keg - 
http://www.google.com/imgres?imgurl=http://pintopinion.files.wordpress.com/2011/05/cartoon-keg.jpg&imgrefurl=http://pintopinion.com/2011/05/13/keg-ride-is-unique-experience-for-bikers/&h=413&w=490&tbnid=dX_Nrxav3p9x4M:&zoom=1&docid=Lahd2gmqrQPV2M&ei=HgEHVfryOIzisASxmIGoAQ&tbm=isch&ved=0CE0QMygUMBQ

Cistern -
http://www.google.com/imgres?imgurl=http://www.gts-mods.com/wp-content/uploads/2012/04/food-and-fuel-cistern-antoine-euro-truck-simulator-2-mod-FoodandFuelCisternAntoine.jpg&imgrefurl=http://www.gts-mods.com/category/german-truck-simulator/gts-trailers/page/40/&h=216&w=460&tbnid=wfEl1lsq75hRlM:&zoom=1&docid=Tj-vruA1-iT6gM&ei=oQEHVabyF8nZsAT9poKQAQ&tbm=isch&ved=0CCMQMygHMAc

**Comment by DinisCruz** [comment]
OK, I'm going to change the values to 1,3,8,13,21,34

One comment I got on the time was that it is 'a bit depressing' if I mark something as 4h and it takes more time for them to do it (so with 1,3,8,13,21,34 what we have is units of work (proportional to who is doing the work))

There are number of [Q] and Taster tasks which are basically small tasks which will take a couple weeks to do (from beginning to end). Usually because there are other dependencies or an answer is needed by somebody else (i.e. aysnc development). The https://github.com/TeamMentor/TM_4_0_Design/issues/701 I just opened up is a good example

And yeah, we can create the convention that above 3, its not really a task

**Comment by DinisCruz** [comment]
ok, they look like this now

![image](https://cloud.githubusercontent.com/assets/656739/7162859/e0560a2a-e38e-11e4-9b97-8e720430c065.png)

lets see how it works in practice

**Comment by DinisCruz** [comment]
Actually, what about this

![image](https://cloud.githubusercontent.com/assets/656739/7162938/47d7f2d0-e38f-11e4-9237-2507118a02a7.png)

Question: which one you prefer, first or second?

**Comment by DinisCruz** [comment]
the latest version even comes our nicely sorted in the labels list :)

![image](https://cloud.githubusercontent.com/assets/656739/7163501/44da7122-e392-11e4-94c1-8f8334e3771c.png)

**Comment by luchia** [comment]
I like it, @DinisCruz.

**Comment by roman87** [comment]
Ok if everyone likes it so be it :)
